src/votekit/ballot_graph.py

```diff
@@ -0,0 +1,226 @@
+import matplotlib.pyplot as plt
+import networkx as nx
+from collections import Counter
+from profile import PreferenceProfile
+from distinctipy import get_colors
+from ballot import Ballot
+from typing import Optional, Callable
+from pydantic import BaseModel, validator
+
+
+def build_graph(n):
```

Comment:
Let's add some doc-strings to this, also is there a reason the live outside the class?

Reply:
Added doc-string. I want to keep it outside so we can use it even if we're not initializing an instance using BallotGraph. Open to alternate suggestions, but I still wanna cache it